File: sdks/python/examples/validate_material_research_bundle.py
from __future__ import annotations

import json
import pathlib
import sys

from kyuubiki_sdk import validate_material_research_bundle


def main() -> None:
    repo_root = pathlib.Path(__file__).resolve().parents[3]
    default_fixture = repo_root / "schemas" / "examples.material-research-bundle.json"
    bundle_path = pathlib.Path(sys.argv[1]) if len(sys.argv) > 1 else default_fixture
    bundle = validate_material_research_bundle(json.loads(bundle_path.read_text()))
    print(f"schema={bundle['schema_version']}")
    print(f"study={bundle['study']}")
    print(f"winner={bundle['summary']['winner_candidate_id']}")
    print(f"reliability={bundle['summary']['reliability_decision']}")
    print(f"next_round={bundle['summary']['next_round_decision']}")
    print(f"next_iteration={bundle['summary'].get('next_iteration', 0)}")
    print(f"runnable_next_steps={bundle['summary'].get('runnable_next_step_count', 0)}")
    print(f"chain_stop={bundle['summary']['chain_stop_reason']}")


if __name__ == "__main__":
    main()

File: sdks/elixir/examples/validate_material_research_bundle.exs
alias KyuubikiSdk.MaterialResearchBundle

default_fixture =
  __DIR__
  |> Path.join("../../../schemas/examples.material-research-bundle.json")
  |> Path.expand()

fixture = System.argv() |> List.first() |> Kernel.||(default_fixture)

{:ok, bundle} =
  fixture
  |> File.read!()
  |> Jason.decode!()
  |> MaterialResearchBundle.validate()

IO.puts("schema=#{bundle["schema_version"]}")
IO.puts("study=#{bundle["study"]}")
IO.puts("winner=#{bundle["summary"]["winner_candidate_id"]}")
IO.puts("reliability=#{bundle["summary"]["reliability_decision"]}")
IO.puts("next_round=#{bundle["summary"]["next_round_decision"]}")
IO.puts("next_iteration=#{bundle["summary"]["next_iteration"] || 0}")
IO.puts("runnable_next_steps=#{bundle["summary"]["runnable_next_step_count"] || 0}")
IO.puts("chain_stop=#{bundle["summary"]["chain_stop_reason"]}")
